frontend/packages/dev-console/src/components/formik-fields/ResourceDropdownField.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import cx from 'classnames';
+import { useField, useFormikContext, FormikValues } from 'formik';
+import { FormGroup } from '@patternfly/react-core';
+import { Firehose, FirehoseResource } from '@console/internal/components/utils';
+import ResourceDropdown from '../dropdown/ResourceDropdown';
+import { DropdownFieldProps } from './field-types';
+import { getFieldId } from './field-utils';
+
+export interface ResourceDropdownFieldProps extends DropdownFieldProps {
+  dataSelector: string[] | number[] | symbol[];
+  resources: FirehoseResource[];
+}
+const ResourceDropdownField: React.FC<ResourceDropdownFieldProps> = ({
+  label,
+  helpText,
+  required,
+  fullWidth,
+  dataSelector,
+  resources,
+  ...props
+}) => {
+  const [field, { touched, error }] = useField(props.name);
+  const { setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
+  const fieldId = getFieldId(props.name, 'ns-dropdown');
+  const isValid = !(touched && error);
+  const errorMessage = !isValid ? error : '';
+  return (
+    <FormGroup
+      fieldId={fieldId}
+      label={label}
+      helperText={helpText}
+      helperTextInvalid={errorMessage}
+      isValid={isValid}
+      isRequired={required}
+    >
+      <Firehose resources={resources}>
+        <ResourceDropdown
+          {...props}
+          id={fieldId}
+          dataSelector={dataSelector}
+          selectedKey={field.value}
+          dropDownClassName={cx({ 'dropdown--full-width': fullWidth })}
+          onChange={(value: string) => {
+            props.onChange && props.onChange(value);
+            setFieldValue(props.name, value);
+            setFieldTouched(props.name, true);
+          }}
+        />
+      </Firehose>
+    </FormGroup>
+  );
+};
+
+export default ResourceDropdownField;
```

Comment:
@christianvogt  i have added this ResourceDropdownField component instead of using NSdropdown ( Which is not highly customizable at the moment and has resource badge next to name and some hardcoded values and classes). So this new formik field will be customizable and can be used with other resources in future.

Reply:
This functions much better now.